packages/core/admin/ee/admin/pages/SettingsPage/pages/ReviewWorkflows/pages/EditView/EditView.js

```diff
@@ -115,21 +115,15 @@ export function ReviewWorkflowsEditView() {
           // changed; this enables partial updates e.g. for users who don't have
           // permissions to see roles
           stages: workflow.stages.map((stage) => {
-            let hasUpdatedPermissions = true;
-            const serverStage = serverState.workflow.stages.find(
-              (serverStage) => serverStage.id === stage?.id
-            );
-
-            if (serverStage) {
-              hasUpdatedPermissions =
-                serverStage.permissions?.length !== stage.permission?.length ||
-                !serverStage.permissions.every(
-                  (serverPermission) =>
-                    !!stage.permissions.find(
-                      (permission) => permission.role === serverPermission.role
-                    )
-                );
-            }
+            const localPermissions = stage?.permissions ?? [];
+            const serverStage = serverState.workflow.stages.find((s) => s.id === stage.id) || {};
+            const serverPermissions = serverStage?.permissions ?? [];
+
+            const hasUpdatedPermissions =
+              localPermissions?.length !== serverPermissions?.length ||
+              localPermissions.some(({ role }) => {
+                return !serverPermissions.some(({ role: serverRole }) => serverRole === role);
+              });
```

Comment:
Why did you have to change this? Asking because this looks a bit different from https://github.com/strapi/strapi/pull/17861/files.

Reply:
You're right I can revert this. Initially I thought this was fixing the partial update bug but rolling it back everything is still working and covered by the BE changes